langchain4j-ollama/src/test/java/dev/langchain4j/model/ollama/OllamaToolChatModelIT.java

```diff
@@ -121,6 +137,58 @@ void should_handle_tool_without_parameter() {
 
     }
 
+    @Test
+    void should_handle_tools_call_in_streaming_scenario() throws Exception {
+        // given
+        UserMessage userMessage = userMessage("What is the weather today in Paris?");
+        List<ToolSpecification> toolSpecifications = singletonList(weatherToolSpecification);
+
+        TestStreamingResponseHandler<AiMessage> handler = new TestStreamingResponseHandler<>();
+
+        // when
+        streamingChatModel.generate(singletonList(userMessage), toolSpecifications, handler);
+
+        Response<AiMessage> aiMessageResponse = handler.get();
+        final AiMessage aiMessage = aiMessageResponse.content();
+
+        // then
+        assertThat(aiMessage.hasToolExecutionRequests()).isTrue();
+        assertThat(aiMessage.toolExecutionRequests().size()).isEqualTo(1);
+        ToolExecutionRequest toolExecutionRequest = aiMessage.toolExecutionRequests().get(0);
+        assertThat(toolExecutionRequest.name()).isEqualTo("get_current_weather");
+        assertThat(toolExecutionRequest.arguments()).isEqualToIgnoringWhitespace("{\"format\": \"celsius\", \"location\": \"Paris\"}");
+
+        // given
+        ToolExecutionResultMessage toolExecutionResultMessage = from(toolExecutionRequest, "{\"format\": \"celsius\", \"location\": \"Paris\", \"temperature\": \"32\"}");
+        List<ChatMessage> messages = asList(userMessage, aiMessage, toolExecutionResultMessage);
+
+        CompletableFuture<Response<AiMessage>> secondFutureResponse = new CompletableFuture<>();
+        streamingChatModel.generate(messages, new StreamingResponseHandler<>() {
+
+            @Override
+            public void onNext(String token) {
+            }
```

Comment:
assertion is needed

Reply:
I've added a counter to check if this method is invoked. Is this what you were thinking of?